### PR TITLE
[rawfile] Add GetFilePos() method

### PR DESCRIPTION
**Pull request by jblomer**
(see also #6458)

### CODE REVIEW

**Review by eguiraud — APPROVED** [review]

File: io/io/inc/ROOT/RRawFile.hxx [hunk 1]
```diff
@@ -160,6 +160,8 @@ public:
    size_t Read(void *buffer, size_t nbytes);
    /// Change the cursor fFilePos
    void Seek(std::uint64_t offset);
+   /// Returns the offset for the next Read/Readln call
+   std::uint64_t GetFilePos() const { return fFilePos; }
```
Comment (pcanal) [inline io/io/inc/ROOT/RRawFile.hxx:164]: Exposing FilePos (and Seek) means than intrinsically RRawFile is thread unsafe.  This is likely 'fine' but we need to make sure that we document it (I.e. a thread safe file accessor would need to be build on to of it)

Comment (eguiraud) [reply io/io/inc/ROOT/RRawFile.hxx:164]: I don't follow..why would users expect that the same RRawFile can be used from multiple threads concurrently? That's never an implied guarantee.

Comment (pcanal) [reply io/io/inc/ROOT/RRawFile.hxx:164]: We have a few request for TFile to be thread-safe.   Not being thread-safe means the client needs to do the lock explicitly (eg. TTree, RDataFrame, etc).  

Eitherway, it never hurts to be explicit :)

Comment (jblomer) [reply io/io/inc/ROOT/RRawFile.hxx:165]: Thanks, clarified in the RRawFile class comment.

**Pull request merged**